testsuite/testsuite-microprofile-jwt/src/test/java/org/wildfly/swarm/microprofile/jwtauth/ApplicationScopedPrincipalLeakTest.java

```diff
@@ -52,26 +52,48 @@ protected static WARArchive initDeployment() {
     public void subjectShouldNotLeakToNonSecuredRequest() throws Exception {
         // project-no-roles-props.yml restricts the number of worker threads to 1,
         // that is, all requests are processed by the same single thread
-
-        String response = Request.Get("http://localhost:8080/mpjwt/subject/secured")
-                .setHeader("Authorization", "Bearer " + createToken("MappedRole"))
-                .execute().returnContent().asString();
-        assertThat(response).isEqualTo(TokenUtils.SUBJECT);
-
```

Comment:
Why do you remove this? The test will no longer verify what it used to verify, no?

Reply:
This is fine, not this test just tests that a null subject is returned for non secure requests, this removed code is now tested  with `subjectFromJsonWebToken()`, let me add 'secureRequest' prefix to some tests.

Reply:
Hmm, those tests don't look like they would check that subject won't leak into a subsequent request...?

Reply:
@Ladicek OK, I've updated all the tests now to check no subsequent request leak occurs